core/aws-core/src/main/java/software/amazon/awssdk/awscore/client/builder/AwsDefaultClientBuilder.java

```diff
@@ -237,6 +240,29 @@ private Boolean dualstackEndpointFromDefaultProvider(SdkClientConfiguration conf
                                        .orElse(null);
     }
 
+    /**
+     * Resolve whether a dualstack endpoint should be used for this client.
+     */
+    private Boolean resolveFipsEndpointEnabled(SdkClientConfiguration config) {
+        return config.option(AwsClientOption.FIPS_ENDPOINT_ENABLED) != null
+               ? config.option(AwsClientOption.FIPS_ENDPOINT_ENABLED)
+               : resolveUseFipsFromDefaultProvider(config);
+    }
+
+    /**
+     * Load the dualstack endpoint setting from the default provider logic.
+     */
+    private Boolean resolveUseFipsFromDefaultProvider(SdkClientConfiguration config) {
+        ProfileFile profileFile = config.option(SdkClientOption.PROFILE_FILE);
+        String profileName = config.option(SdkClientOption.PROFILE_NAME);
+        return FipsEnabledProvider.builder()
+                                  .profileFile(() -> profileFile)
+                                  .profileName(profileName)
+                                  .build()
+                                  .isFipsEnabled()
+                                  .orElse(null);
```

Comment:
Is there any reason we defer to `DefaultServiceEndpointBuilder` to resolve the default value?

Reply:
Old modules that don't currently specify those fields. It's for backwards compatibility.

Reply:
If you're asking why we use null here instead of false: it's so that service-specific config like S3ControlConfiguration resolution logic can differentiate between the customer setting false, and the default value for fips/dualstack being false.